sdk/schemaregistry/schema-registry-avro/src/schemaRegistryAvroSerializer.ts

```diff
@@ -167,6 +167,10 @@ export class SchemaRegistryAvroSerializer {
     }
 
     const schemaResponse = await this.registry.getSchemaById(schemaId);
+    if (!schemaResponse) {
+      throw new Error(`Schema with ID '${schemaId}' not found`);
```

Comment:
could possibly use `RangeError` for this

Reply:
If this is a common scenario that would need to be handled programmatically, we could make a service-specific error with a custom name that devs can look for instead of having to introspect the message

Reply:
My gut feeling is that this isn't an error you would want to handle. I think your architecture should be such that you take the necessary steps to have the needed schema in the registry before trying to serialize/deserialize and failure to do so could be considered a mistake by the caller and not something they should handle, or handle any differently from any other failure to serialize/deserialize (apart from logging the different specifics in the messages somewhere).

But I'm not 100% sure on this. @dhoepelman, do you also use the avro serializer package? What would you like to see from it when the schema can't be matched in the registry?

Reply:
Thank you for confirming!